doc/src/tutorials/physics/duffing-example.rst

```diff
@@ -0,0 +1,183 @@
+.. figure:: duffing-oscillator-pendulum.svg
+
+The system will be modeled using Lagrange equations. `M` is mass of the Duffing oscillator,
+`m` is mass of the pendulum, `l` is length of the pendulum. `k_1` and `k_2` are linear and
+non-linear parts of spring stiffness, and `c_1` is a viscous damping coefficient of the Duffing oscillator.
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> import sympy as sm
+   >>> import sympy.physics.mechanics as me
+   >>> from sympy.physics.mechanics.actuator import DuffingSpring
+   >>> from sympy.physics.mechanics.lagrange import LagrangesMethod
+   >>> from sympy.physics.mechanics.pathway import LinearPathway
+
+Define Variables
+================
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> M, m, l, k1, k2, c1, g = sm.symbols('M, m, l, k1, k2, c1, g')
+   >>> q1, q2 = me.dynamicsymbols('q1 q2')
+   >>> q1d, q2d = me.dynamicsymbols('q1 q2', 1)
+   >>> u1, u2 = me.dynamicsymbols('u1 u2')
+   >>> u1d, u2d = me.dynamicsymbols('u1 u2', 1)
+
+- :math:`q1`: Generalized coordinate representing the position of the Duffing oscillator
+- :math:`q2`: Generalized coordinate representing the angle of the pendulum
+- :math:`q1d`: First time derivative of `q1`, representing the velocity of the Duffing oscillator
+- :math:`q2d`: First time derivative of `q2`, representing the angular velocity of the pendulum
+- :math:`u1`: Generalized speed associated with the Duffing oscillator
+- :math:`u2`: Generalized speed associated with the pendulum
+- :math:`u1d`: First time derivative of `u1`, representing the acceleration of the Duffing oscillator
+- :math:`u2d`: First time derivative of `u2`, representing the angular acceleration of the pendulum
+
+Define Kinematics
+=================
+
+Define all the reference frames and points.
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> # Define reference frames
+   >>> N = me.ReferenceFrame('N')
+   >>> A = N # Duffing Oscillator Block
+   >>> B = N.orientnew('B', 'axis', (q2, N.z)) # Pendulum
+
+The angular velocity of the pendulum in the reference frame is:
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> B.set_ang_vel(N, q2.diff() * N.z)
+
+Locations and velocities of the Duffing Oscillator block and the pendulum are:
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> O = me.Point('O') # fixed point in the inertial reference frame
+   >>> Block = O.locatenew('Block', q1 * N.y)
+   >>> Pendulum = Block.locatenew('Pendulum', l * B.y)
+
+   >>> O.set_vel(N, 0)
+   >>> Block.set_vel(N, u1 * N.y)
+   >>> Pendulum.v2pt_theory(Block, N, B)
+   u1(t)*N.y - l*u2(t)*B.x
+
+   >>> ParBlock = me.Particle('ParBlock', Block, M)
+   >>> ParPendulum = me.Particle('ParPendulum', Pendulum, m)
```

Comment:
Try to follow pep8 for variable names: `par_block`. In mechanics examples we don't often follow it when using single letter names for things, i.e. `A = ReferenceFrame('A')` so that is an exception to the rule.